src/components/CodeEditorWindow/Menu/CommandsDrawer/CommandsDrawer.jsx

```diff
@@ -8,6 +8,19 @@ import useMediaQuery from '@mui/material/useMediaQuery';
 import { useSnackbar } from 'notistack';
 import { getSnackbarOptions } from '../../../Common/utils/snackbarOptions';
 
+const resolveRequiredParams = function (openapi, requestBodyContent) {
+  const contentType = Object.keys(requestBodyContent)[0];
+  if ('$ref' in requestBodyContent[contentType].schema) {
+    const refPath = requestBodyContent[contentType].schema.$ref;
+    // parse and navigate to the ref
+    const pathComponents = refPath.slice(2).split('/');
+    const schema = pathComponents.reduce((doc, pathComponent) => doc[pathComponent], openapi);
+    return schema.required;
+  } else {
+    return requestBodyContent[contentType].schema.required;
+  }
+};
+
```

Comment:
@kartik-gupta-ij Should I move this function to some other utils file?

Reply:
yeah sure!

Reply:
@kartik-gupta-ij I created a `CommandsDrawerUtils` file and put this function there. is that okay?

Reply:
Seems good to me!